src/SSHDebugPS/Docker/DockerConnection.cs

```diff
@@ -242,5 +244,105 @@ private ICommandRunner GetCommandRunner(DockerContainerTransportSettings setting
             else
                 return new RemoteCommandRunner(settings, OuterConnection);
         }
+
+        private const string dockerVersionCommand = "version";
+        private const string dockerVersionArgs = "--format {{.Server.Os}}";
+
+        private const string dockerInfoCommand = "info";
+        private const string dockerInfoArgs = "--format {{.Driver}}";
+
+        public override List<Process> ListProcesses()
+        {
+            string username;
+            TryGetUsername(out username);
+
+            List<Process> processes;
+            string psErrorMessage;
+            // Try using 'ps' first
+            if (!PSListProcess(username, out psErrorMessage, out processes))
```

Comment:
How about the case where PSListProcess fails but ProcFSListProcess succeeded? Might want to handle showing `psErrorMessage`.